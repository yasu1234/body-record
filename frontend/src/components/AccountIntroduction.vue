<script setup>
import { useRouter } from 'vue-router'
import axios from 'axios'
import Cookies from 'js-cookie'

const router = useRouter();

function showSignup() {
    router.push('/signup');
};

function showLogin() {
    router.push('/login');
};

const guestLogin = async () => {
  try {
    const res = await axios.post(import.meta.env.VITE_APP_API_BASE + '/api/v1/users/guest_sign_in')
    Cookies.set('accessToken', response.headers["access-token"])
    Cookies.set('client', response.headers["client"])
    Cookies.set('uid', response.headers["uid"])
  } catch (error) {
    console.log({ error })
  }
}
</script>

<template>
    <h1 class="homeTitle">理想の体を手に入れよう！</h1>
    <div class="homeContents">
        <div class="homeImage">
            <img src="../assets/image/home_image.jpg" alt="Logo" class= "homeLogo">
        </div>
        <div class="homeButtons">
            <button class="commonButton" @click="showSignup">会員登録</button>
            <button class="commonButton" @click="showLogin">ログイン</button>
            <button class="guestButton" @click="guestLogin">ゲストログイン</button>
        </div>    
    </div>
</template>

<style>
.homeContents {
    display: flex;
}
.homeLogo {      
    width:100%;
    height:auto;
    margin-top: 20px;
    object-fit: cover;
    display: block;
}
.homeTitle {      
    text-align: center;
    padding-top:10px;
}
.homeButtons {
    width:50%;
    display:flex;
    flex-direction: column;
    /* margin-left: auto;
    margin-right: auto; */
    align-items: center; /* 水平方向の中央揃え */
    margin: 0 auto; /* 要素を水平方向の中央に配置 */
}
.homeButtons button {
    width: 80%;
    margin-top: 20px;
    font-weight: bold;
    font-size: 18px;
}
.commonButton {
    background-color: #ffa500;
    font-weight: bold;
    font-size: 18px;
}
.guestButton {
    background-color: #f75549;
    font-weight: bold;
    font-size: 18px;
}
.homeImage {
    display:inline-block;
    width:50%;
}
</style>
