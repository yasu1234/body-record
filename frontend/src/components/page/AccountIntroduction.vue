<script setup>
import { useRouter } from "vue-router";
import Cookies from "js-cookie";
import { useToast } from "primevue/usetoast";
import { toastService } from "../../const/toast.js";
import Toast from "primevue/toast";
import axiosInstance from "../../const/axios.js";

const router = useRouter();
const toast = useToast();
const toastNotifications = new toastService(toast);

const showSignup = () => {
  router.push({ name: "Signup" });
};

const showLogin = () => {
  router.push({ name: "Login" });
};

const guestLogin = async () => {
  try {
    const res = await axiosInstance.post("/api/v1/users/guest_sign_in");
    Cookies.set("accessToken", res.headers["access-token"]);
    Cookies.set("client", res.headers["client"]);
    Cookies.set("uid", res.headers["uid"]);

    router.push({ name: "Home" });
  } catch (error) {
    let errorMessages = "";
    if (error.response.status === 422) {
      if (Array.isArray(error.response.data.errors)) {
        errorMessages += error.response.data.errors.join("\n");
      }
    }
    toastNotifications.displayError(
      "ゲストログインに失敗しました",
      errorMessages
    );
  }
};
</script>

<template>
  <Toast position="top-center" />
  <img src="../../assets/image/home_image.jpg" alt="Logo" class="home-logo" />
  <h1 class="text-center mt-5">In-body.comへようこそ！</h1>
  <div class="account-buttons mt-5">
    <button class="account-button" @click="showSignup">会員登録</button>
    <button class="account-button" @click="showLogin">ログイン</button>
    <button class="guest-login-button" @click="guestLogin">
      ゲストログイン
    </button>
  </div>
</template>

<style scoped>
.home-logo {
  width: 100%;
  height: 400px;
  object-fit: cover;
  display: block;
}
.account-buttons {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 20px;
}
.account-buttons button {
  width: 70%;
  font-weight: bold;
  font-size: 18px;
}
.account-button {
  font-weight: bold;
  font-size: 18px;
  padding: 10px 50px;
}
.guest-login-button {
  background-color: #f75549;
  font-weight: bold;
  font-size: 18px;
  padding: 10px 50px;
}

@media screen and (max-width: 768px) {
  .home-logo {
    width: 100%;
    height: 200px;
    object-fit: cover;
    display: block;
  }
}
</style>
