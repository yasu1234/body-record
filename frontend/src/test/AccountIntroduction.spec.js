import { shallowMount } from "@vue/test-utils";
import AccountIntroduction from "@/components/page/account/AccountIntroduction.vue";

jest.mock("axios");
jest.mock("vue-router", () => {
  const router = {
    push: jest.fn(),
    pathname: "/",
    route: "",
    asPath: "",
    query: {},
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
  };
  return {
    useRouter: jest.fn().mockReturnValue(router),
    useRoute: jest.fn().mockReturnValue({
      params: {
        id: 1,
      },
    }),
  };
});

jest.mock("primevue/usetoast", () => ({
  useToast: jest.fn(() => ({
    add: jest.fn(),
    removeAll: jest.fn(),
  })),
}));

jest.mock("@/js/axios.js", () => ({
  axiosInstance: {
    get: jest.fn(),
    interceptors: {
      request: { use: jest.fn() },
      response: { use: jest.fn() },
    },
  },
  setupInterceptors: jest.fn(),
}));

describe("Contact", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(AccountIntroduction, {
      global: {
        stubs: ["router-link", "router-view"],
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("SnapShotテスト初期表示", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
