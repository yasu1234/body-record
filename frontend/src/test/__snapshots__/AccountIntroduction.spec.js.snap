// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact SnapShotテスト初期表示 1`] = `
<div
  data-v-app=""
>
  
  <toast-stub
    position="top-center"
  />
  <img
    alt="Logo"
    class="home-logo"
    src=""
  />
  <p
    class="introduction-title mt-5"
  >
    In-body.comへようこそ！
  </p>
  <div
    class="account-buttons mt-5"
  >
    <button
      class="account-button"
    >
      会員登録
    </button>
    <button
      class="account-button"
    >
      ログイン
    </button>
    <button
      class="guest-login-button"
    >
       ゲストログイン 
    </button>
  </div>
  
</div>
`;
